### PR TITLE
router batching stable commit 1

### PR DESCRIPTION
**Pull request by chandumlg**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: router/router.go [hunk 1]
```diff
@@ -137,8 +140,100 @@ func (rt *HandleT) trackStuckDelivery() chan struct{} {
...
+
+			destinationJobParametersArray := []DestinationJobParametersT{{UserID: userID, JobID: job.JobID, SourceID: parameters.SourceID, DestinationID: parameters.DestinationID, AttemptNum: job.LastJobStatus.AttemptNum, ReceivedAt: parameters.ReceivedAt, CreatedAt: job.CreatedAt}}
+			destinationJob := &DestinationJobT{Payload: job.EventPayload, JobParametersArray: destinationJobParametersArray}
+			worker.destinationJobs = append(worker.destinationJobs, destinationJob)
+
+			if rt.enableBatching {
+				if len(worker.destinationJobs) == noOfJobsToBatchInAWorker {
+					worker.destinationJobs = rt.batch(worker.destinationJobs)
+					rt.handleWorkerDestinationJobs(worker)
+					worker.destinationJobs = nil
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply router/router.go:253]: Removing reference to the old slice. I don't think it hurts.

File: router/router.go [hunk 2]
```diff
@@ -137,8 +140,100 @@ func (rt *HandleT) trackStuckDelivery() chan struct{} {
...
+				}
+			} else {
+				rt.handleWorkerDestinationJobs(worker)
+				worker.destinationJobs = nil
+				worker.destinationJobs = make([]*DestinationJobT, 0)
+			}
+
+		case <-timeout:
+			timeout = time.After(jobsBatchTimeout)
+			if len(worker.destinationJobs) > 0 {
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply router/router.go]: I felt its not necessary. But can be done. I don't have an opinion here.

File: config/config.toml [hunk 3]
```diff
@@ -156,6 +158,7 @@ minRetryBackoffInS = 10
 maxRetryBackoffInS = 300
     [Router.BRAZE]
     forceHTTP1=true
```
[the remark this answers is not in the document]
Comment (chandumlg) [reply config/config.toml:160]: After transformer changes for BRAZE are done, we can do this reproducing and removing forceHTTP1 stuff.

[the remark this answers is not in the document]
Comment (snarkychef) [reply config/config.toml:160]: Got it.

File: router/router.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (snarkychef) [reply router/router.go:253]: Reference will be removed automatically right? This becomes a practise and affects our understanding.

File: router/router.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (snarkychef) [reply router/router.go]: We can leave it then.

**Review by snarkychef — APPROVED** [review]

**Pull request merged**